bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/tracer/BSpan.java

```diff
@@ -76,52 +79,74 @@ public static BSpan start(String serviceName, String operationName, boolean isCl
      */
     public static BSpan start(BSpan parentSpan, String serviceName, String operationName, boolean isClient) {
         Tracer tracer = TracersStore.getInstance().getTracer(serviceName);
-        return start(tracer, parentSpan.span.context(), operationName, isClient);
+        Context parentContext = Context.current().with(parentSpan.span);
+        return start(tracer, parentContext, operationName, isClient);
     }
 
     /**
      * Start a new span with a parent using parent trace context.
      * The started span is part of a trace which had spanned across multiple services and the parent is in the service
      * which called the current service.
      *
-     * @param parentTraceContext The parent trace context
-     * @param serviceName        The name of the service the span belongs to
-     * @param operationName      The name of the operation the span corresponds to
-     * @param isClient           True if this is a client span
+     * @param httpHeaders   Contains http headers of request received
+     * @param serviceName   The name of the service the span belongs to
+     * @param operationName The name of the operation the span corresponds to
+     * @param isClient      True if this is a client span
      * @return The new span
      */
-    public static BSpan start(Map<String, String> parentTraceContext, String serviceName, String operationName,
+    public static BSpan start(Map<String, String> httpHeaders, String serviceName, String operationName,
```

Comment:
Let's not use the word http.
```suggestion
    public static BSpan start(Map<String, String> parentTraceContext, String serviceName, String operationName,
```

Reply:
done